### PR TITLE
ipsec: Remove deprecated upgrade code

### PR DESCRIPTION
**Pull request by pchaigno**
The first commit removes support for the global-key system in IPsec, which had been deprecated in v1.16. The second commit removes IPsec upgrade logic that isn't needed anymore. The third commit fixes the unit tests to support the per-tunnel-keys system.

```release-note
Remove support for the insecure, deprecated global IPsec key. Per-tunnel IPsec keys will now be used regardless of the IPsec secret format.
```

### CODE REVIEW

File: .github/actions/ipsec-key-rotate/action.yaml [hunk 1]
```diff
@@ -5,18 +5,18 @@ inputs:
...
     type: string
     description: "gcm(aes) or cbc(aes)"
-  key-type-one:
+  encryption-overlay:
     required: true
     type: string
-    description: "'+' if we started with the per-tunnel key system"
-  key-type-two:
+    description: "'true' if encryption-overlay is enabled"
+  nb-nodes:
```
Comment (aanm) [inline .github/actions/ipsec-key-rotate/action.yaml]: `nb-nodes` is not clear enough without making the user to go to the documentation try to understand what `nb` means.

```suggestion
  number-nodes:
```

File: .github/workflows/conformance-eks.yaml [hunk 2]
```diff
@@ -360,8 +360,8 @@ jobs:
         uses: ./.github/actions/ipsec-key-rotate
         with:
           key-algo: "gcm(aes)"
-          key-type-one: ""
-          key-type-two: ""
+          nb-nodes: 3
```
Comment (aanm) [inline .github/workflows/conformance-eks.yaml]:
```suggestion
          number-nodes: 3
```

**Review by aanm — DISMISSED** [review]

File: .github/workflows/conformance-ipsec-e2e.yaml [hunk 3]
```diff
@@ -306,7 +290,7 @@ jobs:
             echo "Invalid key type"; exit 1
           fi
           kubectl create -n kube-system secret generic cilium-ipsec-keys \
-            --from-literal=keys="3${{ matrix.key-type-one }} ${key}"
+            --from-literal=keys="3+ ${key}"
```
Comment (julianwiedmann) [inline .github/workflows/conformance-ipsec-e2e.yaml]: Can't we just phase out the `+` symbol instead (except in downgrade tests) ?

And have some unit tests to confirm that `parseSPI()` still has toleration for both styles of secret string.

**Review by julianwiedmann — COMMENTED** [review]
Just one question, the change looks good!

**Review by qmonnet — APPROVED** [review]
Looks good from my side. Nice clean-up!

**Review by thorn3r — APPROVED** [review]
LGTM for Clustermesh changes

**Review by tklauser — APPROVED** [review]

File: .github/workflows/tests-clustermesh-upgrade.yaml [hunk 4]
```diff
@@ -286,7 +286,7 @@ jobs:
       - name: Create the IPSec secret in both clusters
         if: matrix.encryption == 'ipsec'
         run: |
-          SECRET="3 rfc4106(gcm(aes)) $(openssl rand -hex 20) 128"
+          SECRET="3+ rfc4106(gcm(aes)) $(openssl rand -hex 20) 128"
```
Comment (julianwiedmann) [inline .github/workflows/tests-clustermesh-upgrade.yaml]: fyi, I switched this over in https://github.com/cilium/cilium/pull/35295.

File: .github/workflows/conformance-ipsec-e2e.yaml (hunk 3 above)
Comment (pchaigno) [reply .github/workflows/conformance-ipsec-e2e.yaml]: > Can't we just phase out the `+` symbol instead (except in downgrade tests) ?

I'm trying to match what users will be running in production for the short and medium term. Given we're not asking to remove the `+`, I'd expect all of them to have it.

**Review by rgo3 — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Pull request merged**